pkg/kubectl/run.go

```diff
@@ -64,7 +66,25 @@ func makePodSpec(params map[string]string, name string) (*api.PodSpec, error) {
 	return &spec, nil
 }
 
-func (BasicReplicationController) Generate(params map[string]string) (runtime.Object, error) {
+func (BasicReplicationController) Generate(genericParams map[string]interface{}) (runtime.Object, error) {
+	args := []string{}
+	val, found := genericParams["args"]
+	if found {
+		var isArray bool
+		args, isArray = val.([]string)
+		if !isArray {
+			return nil, fmt.Errorf("expected []string, found: %v", val)
+		}
+		delete(genericParams, "args")
```

Comment:
given this is a magic key, couldn't you just as easily do an `if/continue` below and avoid the mutation?